Corona-Warn-App/src/main/java/de/rki/coronawarnapp/vaccination/ui/list/VaccinationListViewModel.kt

```diff
@@ -0,0 +1,203 @@
+                )
+            )
+        )
+        val nameDataProofCertificate = ProofCertificateV1.NameData(
+            givenName = "François-Joan",
+            givenNameStandardized = "FRANCOIS<JOAN",
+            familyName = "d'Arsøns - van Halen",
+            familyNameStandardized = "DARSONS<VAN<HALEN",
+        )
+        val proofCertificates = setOf(
+            ProofCertificateV1(
+                version = "1.0.0",
+                nameData = nameDataProofCertificate,
+                dateOfBirth = dateOfBirth,
+                vaccinationDatas = listOf(
+                    ProofCertificateV1.VaccinationData(
+                        targetId = "840539006",
+                        vaccineId = "1119349007",
+                        medicalProductId = medicalProductId,
+                        marketAuthorizationHolderId = "ORG-100030215",
+                        doseNumber = 2,
+                        totalSeriesOfDoses = 2,
+                        vaccinatedAt = LocalDate.parse("2021-04-22"),
+                        countryOfVaccination = "DE",
+                        certificateIssuer = "Ministry of Public Health, Welfare and Sport",
+                        uniqueCertificateIdentifier = "urn:uvci:01:NL:THECAKEISALIE",
+                    )
+                )
+            )
+        )
+
+        val listItems = assembleItemList(
+            vaccinationCertificates = vaccinationCertificates,
+            proofCertificates = proofCertificates,
+            firstName = "François-Joan",
+            lastName = "d'Arsøns - van Halen",
+            dateOfBirth = dateOfBirth,
+            vaccinationStatus,
+            timeStamper.nowUTC.plus(Duration.standardDays(3))
+        )
+
+        UiState(listItems, vaccinationStatus = vaccinationStatus)
+    }.catch {
+        // TODO Error Handling in an upcoming subtask
+    }.asLiveData()
+
+    // TODO: after using actual values from the repository, we only pass VaccinatedPerson here instead of all these
+    // arguments
+    @Suppress("LongParameterList")
+    private fun assembleItemList(
+        vaccinationCertificates: Set<VaccinationCertificateV1>,
+        proofCertificates: Set<ProofCertificateV1>,
+        firstName: String,
+        lastName: String,
+        dateOfBirth: LocalDate,
+        vaccinationStatus: VaccinatedPerson.Status,
+        proofExpiresAt: Instant
+
+    ) = mutableListOf<VaccinationListItem>().apply {
+        if (vaccinationStatus == COMPLETE) {
+            if (proofCertificates.isNotEmpty()) {
+
+                val proofCertificate = proofCertificates.first()
+                val expiresAt = proofExpiresAt.toLocalDateUserTz()
+                val today = timeStamper.nowUTC.toLocalDateUserTz()
+                val remainingValidityInDays = Days.daysBetween(today, expiresAt).days
+
+                add(
+                    VaccinationListCertificateCardItem(
+                        qrCode = null, // TODO: Generate QR-code
```

Comment:
I think you can use QrCode generator and pass any string for now `qrcodeGenerator.generateQrCode("TODO")`

Reply:
Good Idea!

I am now using `qrCodeGenerator` together with `viewModelScope` in `VaccinationListCertificateCardItemVH` so that QR-Code generation doesn't block the other cards from being loaded. Not sure if this is a good idea 🤔

Reply:
Implemented the suggestion from @d4rken . See below.